### PR TITLE
Update google calendar documentation for yaml deprecation

### PR DESCRIPTION
**Pull request by allenporter**
## Proposed change

Update google calendar documentation for yaml deprecation with migration of oauth, entity disabling, and oauth access. This also de-emphasises the `google_calendars.yaml` which has not been deprecated yet, but is discouraged as we move more configuration to the UI in preparation for supporting multiple config entries.

This also adds some high level overview about automations and links to the primary calendar automation documentation for more details.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [X] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/72288
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [X] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [X] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/google.markdown [hunk 1]
```diff
@@ -38,69 +38,14 @@ This section explains how to generate a Client ID and Client Secret on
...
+1. You will then be presented with a pop-up saying 'OAuth client created' showing `Your Client ID` and `Your Client Scret`. Make a note of these (for example, copy and paste them into a text editor) as you will need to put these in your `configuration.yaml` file shortly. Once you have noted these strings, click `OK`. If you need to find these credentials again at any point then simply navigate to `APIs & Services` > `Credentials` and you will see `Home Assistant Credentials` (or whatever you named them in the previous step) under `OAuth 2.0 Clident IDs`. To view both the `Client ID` and `Client secret`, click on the pencil icon, this will take you to the settings page for these credentials and the information will be on the right hand side of the page.
```
Comment (MartinHjelmare) [inline source/_integrations/google.markdown]: Does it really say `Scret`?

File: source/_integrations/google.markdown [hunk 2]
```diff
@@ -136,49 +81,83 @@ to all the Google Calendars available to your linked account.
...
+See [Calendar Automations](/integrations/calendar#automation) for an overview, and read more about [Calendar Trigger Variables](/docs/automation/templating/#calendar) for the available information you can use in a condition or action such as the event `summary`, `description`, `location` and more.
 
-A bit more elaborate configuration:
+## Calendar Entity Attributes
+
+The calendar entity has additional attributes related to a single next upcoming event.
+
+<div class='note'>
+
+Using the entity state and attributes is more error prone and less flexible than using Calendar Automations. The calendar entity itself may only tracks a single upcoming active event and can't handle multiple events with the same start time, or overlapping events.
```
Comment (MartinHjelmare) [inline source/_integrations/google.markdown]:
```suggestion
Using the entity state and attributes is more error prone and less flexible than using Calendar Automations. The calendar entity itself may only track a single upcoming active event and can't handle multiple events with the same start time, or overlapping events.
```

File: source/_integrations/google.markdown [hunk 3]
```diff
@@ -136,49 +81,83 @@ to all the Google Calendars available to your linked account.
...
 
 </div>
 
+{% enddetails %}
+
+## More Configuration
+
+{% details "More Configuration" %}
+
+The integration will discover new calendars and write them into a configuration file `google_calendars.yaml` in your configuration directory that can allow for additional configuration. It is not recommended to new users to use these settings as they are not
```
Comment (MartinHjelmare) [inline source/_integrations/google.markdown]: Should the sentence that discourages using this feature be a warning? Does that work within a detail?

File: source/_integrations/google.markdown (hunk 1 above)
Comment (allenporter) [reply source/_integrations/google.markdown]: No, reverting mistake.

File: source/_integrations/google.markdown (hunk 3 above)
Comment (allenporter) [reply source/_integrations/google.markdown]: Looks like it works. I've split the message into two parts so the warning is a little smaller and easier to read.

Comment (allenporter) [reply source/_integrations/google.markdown]: One other thing is that i've started adding some recipes to the `calendar` page https://github.com/home-assistant/home-assistant.io/pull/22846 to try to help users understand how to get some of the functionality they get with what they believe is a simple binary sensor.

Maybe it makes sense to move that here...

Comment (MartinHjelmare) [reply source/_integrations/google.markdown]: I think it's better to link to a single page with examples than to add examples to multiple pages.

**Review by MartinHjelmare — APPROVED** [review]

**Pull request merged**